### DISCUSSION TITLE
Are scheduler tasks still active when ShutdownEvent is fired

### DISCUSSION THREAD

**Discussion by bulldog98** [quarkusio/quarkus#34807] [Q&A] [answered]
I have a function like so
```java
void onShutdown(final @Observes ShutdownEvent ev) {
    // cleanly close some communication
}
```
And I have a scheduler that updates the communication periodically.

My Question can the scheduled task (function) still be called while the `onShutdown` is already executed?

[a comment by an automated account is not shown]

**Comment by mkouba** [comment, marked as answer]
Nope, this is not guaranteed. 

In general, the `ShutdownEvent` is fired before the CDI container (ArC) is shut down so that a bean can observe this event and the scheduler (either the impl from `quarkus-scheduler` or from `quarkus-quartz`) is shut down after the CDI application/singleton contexts are destroyed. Moreover, a scheduled task can be running during the shutdown.

Finally, if the [`quarkus.quartz.shutdown-wait-time`](https://quarkus.io/guides/quartz#quarkus-quartz_quarkus.quartz.shutdown-wait-time) config property is set to a non-zero value, then Quarkus will attempt to wait for all running jobs to finish (`quarkus-quartz` only).
